libjulia/backends/evm/EVMCodeTransform.cpp

```diff
@@ -546,10 +659,12 @@ void CodeTransform::checkStackHeight(void const* _astElement) const
 {
 	solAssert(m_info.stackHeightInfo.count(_astElement), "Stack height for AST element not found.");
 	solAssert(
-		m_info.stackHeightInfo.at(_astElement) == m_assembly.stackHeight() - m_stackAdjustment,
+		m_info.stackHeightInfo.at(_astElement) == static_cast<int>(m_reusedVariables.size()) + m_assembly.stackHeight() - m_stackAdjustment,
```

Comment:
This function reuses some of these variables and it seems quite unreadable now. Can you refactor this function a bit? Maybe it makes even sense doing that in a separate PR.

Reply:
submitted a new PR https://github.com/ethereum/solidity/pull/5093